src/components/set-image-card/SetImageCard.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useEffect, useRef, useState } from "react";
+
+import { FlatActionButton } from "../flat-action-button";
+import { Button } from "../button";
+import { Divider } from "../divider";
+import { IconDelete, IconEdit } from "../../icons";
+import { Img } from "../../base-components";
+
+import { SSetImageCard } from "./styles/SSetImageCard";
+import { SSetImageCardBody } from "./styles/SSetImageCardBody";
+import { SSetImageCardButtonsBar } from "./styles/SSetImageCardButtonsBar";
+import { SSetImageCardPreview } from "./styles/SSetImageCardPreview";
+import { SSetImageCardTitle } from "./styles/SSetImageCardTitle";
+import { SSetImageCardSubTitle } from "./styles/SSetImageCardSubTitle";
+
+interface ISetImageCard {
+  onChange: (url: string) => void;
+  processImage: (file: File) => Promise<string>;
+  initalValue?: string;
+}
+
+export const SetImageCard: React.FC<ISetImageCard> = ({ processImage, onChange, initalValue }) => {
+  const [image, setImage] = useState<string | null>(initalValue || null);
+  const [isLoading, setLoading] = useState<boolean>(false);
+
+  const imageRef = useRef<HTMLInputElement>(null);
+
+  const mockProcessIamge = () => {
```

Comment:
this function name `mockProcessIamge` is a bit off to me ^^ - i would rather name it `triggerImageSelection`
or something

Reply:
you're right! forgot this one, will `openImagePicker` work?